modules/sublimeBidAdapter.js

```diff
@@ -1,135 +1,282 @@
+ *
+ * @param {BidRequest} bid The bid params to validate.
+ * @return {Boolean} True if this is a valid bid, and false otherwise.
+ */
+function isBidRequestValid(bid) {
+  return !!Number(bid.params.zoneId);
+}
+
+/**
+ * Make a server request from the list of BidRequests.
+ *
+ * @param {BidRequest[]} validBidRequests An array of bids
+ * @param {Object} bidderRequest - Info describing the request to the server.
+ * @return ServerRequest Info describing the request to the server.
+ */
+function buildRequests(validBidRequests, bidderRequest) {
+  window.sublime = window.sublime || {};
+
+  let commonPayload = {
+    sublimeVersion: SUBLIME_VERSION,
+    // Current Prebid params
+    prebidVersion: '$prebid.version$',
+    currencyCode: config.getConfig('currency.adServerCurrency') || DEFAULT_CURRENCY,
+    timeout: typeof bidderRequest !== 'undefined' ? bidderRequest.timeout : config.getConfig('bidderTimeout'),
+  };
+
+  // RefererInfo
+  if (bidderRequest && bidderRequest.refererInfo) {
+    commonPayload.referer = bidderRequest.refererInfo.referer;
+    commonPayload.numIframes = bidderRequest.refererInfo.numIframes;
+  }
+  // GDPR handling
+  if (bidderRequest && bidderRequest.gdprConsent) {
+    commonPayload.gdprConsent = bidderRequest.gdprConsent.consentString;
+    commonPayload.gdpr = bidderRequest.gdprConsent.gdprApplies; // we're handling the undefined case server side
+
+    // Injecting gdpr consent into sublime tag
+    window.sublime.gdpr = window.sublime.gdpr || {};
+    window.sublime.gdpr.injected = {
+      consentString: bidderRequest.gdprConsent.consentString,
+      gdprApplies: bidderRequest.gdprConsent.gdprApplies
+    };
+  }
+
+  return validBidRequests.map(bid => {
+    let bidHost = bid.params.bidHost || DEFAULT_BID_HOST;
+    let callbackName = (bid.params.callbackName || DEFAULT_CALLBACK_NAME) + '_' + bid.params.zoneId;
+    let protocol = bid.params.protocol || DEFAULT_PROTOCOL;
+    let sacHost = bid.params.sacHost || DEFAULT_SAC_HOST;
+
+    setState({
+      transactionId: bid.transactionId,
+      zoneId: bid.params.zoneId,
+      debug: bid.params.debug || false,
     });
-  },
-
-  /**
-     * Unpack the response from the server into a list of bids.
-     *
-     * @param {*} serverResponse A successful response from the server.
-     * @param {*} bidRequest An object with bid request informations
-     * @return {Bid[]} An array of bids which were nested inside the server.
-     */
-  interpretResponse: (serverResponse, bidRequest) => {
-    const bidResponses = [];
-    const response = serverResponse.body;
-
-    if (response) {
-      if (response.timeout || !response.ad || response.ad.match(/<!-- No ad -->/gmi)) {
-        return bidResponses;
-      }
 
-      // Setting our returned sizes object to default values
-      let returnedSizes = {
-        width: 1800,
-        height: 1000
-      };
+    // Adding Sublime tag
+    let script = document.createElement('script');
```

Comment:
If script is needed to be _sync_ : there is no guarantee that code is executed _after_ it is loaded. Is it the required behavior?

If _not_, then I would add
`script.async = true`
...to let the browser not blocking the whole page parsing.

Reply:
We could test if async is more performante for this implementation in an other technical story!